From 46284e5adda90a547599520193dfbca6cbd09573 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 15:30:10 -0400
Subject: [block] mtip32xx: fix bad use of smp_processor_id()

Message-id: <1402673421-17117-8-git-send-email-snitzer@redhat.com>
Patchwork-id: 83917
O-Subject: [RHEL7.1 PATCH 07/18] mtip32xx: fix bad use of smp_processor_id()
Bugzilla: 1105703
RH-Acked-by: David Milburn <dmilburn@redhat.com>

BZ: 1105703

Upstream commit 7f328908f9cb69a72ce1c9279508c786cf85f1f5
Author: Jens Axboe <axboe@fb.com>
Date:   Mon Mar 10 14:29:37 2014 -0600

    mtip32xx: fix bad use of smp_processor_id()

    mtip_pci_probe() dumps the current CPU when loaded, but it does
    so in a preemptible context. Hence smp_processor_id() correctly
    warns:

    BUG: using smp_processor_id() in preemptible [00000000] code: systemd-udevd/155
    caller is mtip_pci_probe+0x53/0x880 [mtip32xx]

    Switch to raw_smp_processor_id(), since it's just informational
    and persistent accuracy isn't important.

    Signed-off-by: Jens Axboe <axboe@fb.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/block/mtip32xx/mtip32xx.c b/drivers/block/mtip32xx/mtip32xx.c
index 8573eb6..643773b 100644
--- a/drivers/block/mtip32xx/mtip32xx.c
+++ b/drivers/block/mtip32xx/mtip32xx.c
@@ -4506,7 +4506,7 @@ static int mtip_pci_probe(struct pci_dev *pdev,
  }
  dev_info(&pdev->dev, "NUMA node %d (closest: %d,%d, probe on %d:%d)\n",
   my_node, pcibus_to_node(pdev->bus), dev_to_node(&pdev->dev),
-  cpu_to_node(smp_processor_id()), smp_processor_id());
+  cpu_to_node(raw_smp_processor_id()), raw_smp_processor_id());
 
  dd = kzalloc_node(sizeof(struct driver_data), GFP_KERNEL, my_node);
  if (dd == NULL) {
-- 
1.7.1